EG-3164: Update release notes for 4.4.3 (#188)
corda/corda-docs: content/en/docs/corda-enterprise/4.4/release-notes-enterprise.md

# Corda Enterprise Release notes

## Corda Enterprise 4.4.3

Corda Enterprise 4.4.3 is a patch release of Corda Enterprise that introduces fixes to known issues in Corda Enterprise 4.4.2.

> 

Reviewer: Missing boilerplate about upgrade? See here for reference: https://github.com/corda/corda-docs/pull/187/files

Reviewer 2: Agreed, please add:

### Upgrade recommendation

As a developer, you should upgrade to the [latest released version of Corda](https://docs.corda.net/docs/corda-enterprise/index.html) as soon as possible. Check the latest Corda Enterprise release notes and upgrade guide [here](https://docs.corda.net/docs/corda-enterprise/release-notes-index.html).

As a node operator, you should upgrade to the [latest released version of Corda](https://docs.corda.net/docs/corda-enterprise/index.html) if the fixed issues listed below are relevant to your work.

### Issues fixed

* Allow session rollbacks inside an entity manager.

> * Database Management Tool - Sensitive information being exposed as clear text on logs and terminal. 

Reviewer: I think these are phrased inconsistently - the list contains both issue descriptions ("Config Obfuscator doesn't work anymore") and actions taken to fix the issues ("Prevent AMQP header spoofing"). We should settle on one format (I prefer the latter but happy to conform to the tech writing stile guide).

Reviewer 2: Agree, these entries are mostly taken from the Jira descriptions but don't sound like. Please reword here and below - see my suggestions in https://github.com/corda/corda-docs/pull/190 regarding the format/style for now.

Author: Descriptions updated.

* Classloader fails to find class when Optional generic used on Type definition.
* Config Obfuscator doesn't work anymore for HSM configuration files.

> * Revert H2 version to 197 to avoid bug introduced in the dependency after.

Reviewer: Actual version is 1.4.197, per `build.gradle` file

* Prevent AMQP header spoofing.

> * Uncaught exception in FlowStateMachineImpl.run().initialiseFlow().

Reviewer 2: add code style (on top of the rewrite)

* NodeMeteringBackground processes high CPU while a node is idle.
* Move backchainFetchBatchSize option into the tuning section.
* Error in DNS name resolution when using Corda Health Survey tool.

> * [ConfigObfuscator] Firewall doesn't start if its main config and HSM config was obfuscated with specified SEED and PASSPHRASE.

Reviewer: This looks like a copy-paste from the ticket description - would get rid of the square brackets and rephrase

# Corda Enterprise Release notes

## Corda Enterprise 4.4.3

Corda Enterprise 4.4.3 is a patch release of Corda Enterprise that introduces fixes to known issues in Corda Enterprise 4.4.2.

### Upgrade recommendation

As a developer, you should upgrade to the [latest released version of Corda](https://docs.corda.net/docs/corda-enterprise/index.html) as soon as possible. Check the latest Corda Enterprise release notes and upgrade guide [here](https://docs.corda.net/docs/corda-enterprise/release-notes-index.html).

As a node operator, you should upgrade to the [latest released version of Corda](https://docs.corda.net/docs/corda-enterprise/index.html) if the fixed issues listed below are relevant to your work.

### Fixed issues

* Session rollbacks are now allwed inside an entity manager.

> * Sensitive information is no longer exposed as clear text on logs and terminal when using the [Database Management Tool](database-management-tool.mc). 

Reviewer 2: ```suggestion
* Sensitive information is no longer exposed as clear text on logs and terminal when using the [Database Management Tool](database-management-tool.md). 
```

> * Classloader failed to find class when Optional generic used on Type definition.

Reviewer 2: ```suggestion
* Fixed an issue where the Classloader failed to find a class when Optional generic was used on Type definition.
```

Reviewer 2: just make sure to clarify on Optional generic / Type definition, and apply code style as needed

> * Classloader failed to find class when Optional generic used on Type definition.

Author: Updated to "Fixed an issue where the Classloader failed to find a Command class when Optional generic was used on Type definition.” (From @mnesbit on Slack)

* Config Obfuscator did not work for HSM configuration files.
* Reverted H2 version to 1.4.197 to avoid a dependency bug introduced after the previous upgrade.

> * Security update to prevent AMQP header spoofing.

Reviewer 2: ```suggestion
* A security update to prevent AMQP header spoofing has been applied.
```

> * Uncaught exception in `FlowStateMachineImpl.run().initialiseFlow()` now correctly handled.

Reviewer 2: ```suggestion
* A previously unhandled exception in `FlowStateMachineImpl.run().initialiseFlow()` is now handled correctly.
```

> * `NodeMeteringBackground` process CPU use has been decreased.

Reviewer 2: ```suggestion
* The CPU usage of the `NodeMeteringBackground` process has been decreased.
```

> * Moved `backchainFetchBatchSize` option into the tuning section of the node configuration file.

Reviewer 2: ```suggestion
* The `backchainFetchBatchSize` option has been moved to the tuning section of the node configuration file.
```

> * Fixed an error in DNS name resolution when using Corda Health Survey tool.

Reviewer 2: ```suggestion
* Fixed an error in DNS name resolution when using the [Corda Health Survey tool](health-survey.md).
```

> * Firewall did not start if its main configuration and HSM configuation was obfuscated.

Reviewer 2: ```suggestion
* Fixed an issue where Corda Firewall did not start if its main configuration and its HSM configuration were obfuscated.
```

> * Deobfuscation options were missing from HA-Utilities in `generate-internal-tunnel-ssl-keystores` mode.

Reviewer 2: ```suggestion
* Fixed an issue where deobfuscation options were missing from HA-Utilities in `generate-internal-tunnel-ssl-keystores` mode.
```

> * Corda Enterprise 4.5 features backported to allow AKS deployment in 4.4.3.

Reviewer 2: ```suggestion
* Some Corda Enterprise 4.5 features have been backported to allow AKS deployment in Corda Enterprise 4.4.3.
```

> * Fixed an issue where deobfuscation options were missing from HA-Utilities in `generate-internal-tunnel-ssl-keystores` mode.

Reviewer 2: ```suggestion
* Fixed an issue where deobfuscation options were missing from [HA Utilities](ha-utilities.md) in `generate-internal-tunnel-ssl-keystores` mode.
```

# Corda Enterprise Release notes

## Corda Enterprise 4.4.3

Corda Enterprise 4.4.3 is a patch release of Corda Enterprise that introduces fixes to known issues in Corda Enterprise 4.4.2.

### Upgrade recommendation

As a developer, you should upgrade to the [latest released version of Corda](https://docs.corda.net/docs/corda-enterprise/index.html) as soon as possible. Check the latest Corda Enterprise release notes and upgrade guide [here](https://docs.corda.net/docs/corda-enterprise/release-notes-index.html).

As a node operator, you should upgrade to the [latest released version of Corda](https://docs.corda.net/docs/corda-enterprise/index.html) if the fixed issues listed below are relevant to your work.

### Fixed issues

> * Session rollbacks are now allwed inside an entity manager.

Reviewer: Typo -> allowed

> * Session rollbacks are now allwed inside an entity manager.

Reviewer 2: ```suggestion
* Session rollbacks are now allowed inside an entity manager.
```
